src/cc/CCtokens_impl.h

```diff
@@ -311,40 +319,63 @@ UniValue TokenTransferExt(const CPubKey &remotepk, CAmount txfee, uint256 tokeni
         CCerror = "mypk is not set or invalid";
         return  NullUniValue;
     }
+    std::vector<CPubKey> destpubkeys;
+    for (auto const &dest : destinations)
+        if (dest.which() == TX_PUBKEY)
+            destpubkeys.push_back(boost::get<CPubKey>(dest));
 
     // CAmount normalInputs = AddNormalinputs(mtx, mypk, txfee, 0x10000, isRemote);   // note: wallet scanning for inputs is slower than index scanning
     CAmount normalInputs = AddNormalinputsRemote(mtx, mypk, txfee, 0x10000, useMempool);
 
     if (normalInputs > 0)
 	{        
-		if ((inputs = AddTokenCCInputs<V>(cp, mtx, tokenaddr, tokenid, total, CC_MAXVINS, useMempool)) >= total)  // NOTE: AddTokenCCInputs might set cp->additionalEvalCode which is used in FinalizeCCtx!
+        CAmount CCchange = 0, CCinputs = 0;  
+        for (const auto &addr : tokenaddrs)  {
+            CAmount CCinputsOne = AddTokenCCInputs<V>(cp, mtx, addr.c_str(), tokenid, total, CC_MAXVINS, useMempool);
+            if (CCinputsOne > 0)
+                CCinputs += CCinputsOne;
+            if (CCinputs >= total)
+                break;
+        }
+
+        if (spendMarker)  {
+            mtx.vin.push_back(CTxIn(tokenid, 0));  // spend tokenlist marker
+        }
+
+		if (CCinputs >= total)  
     	{  
-			if (inputs > total)
-				CCchange = (inputs - total);
+			if (CCinputs > total)
+				CCchange = (CCinputs - total);
 
-            if (destpubkeys.size() == 0) {
+            if (destinations.size() == 0) {
                 CCerror = "no dest pubkeys";
                 return NullUniValue;
             }
 
             if (V::EvalCode() == EVAL_TOKENS) {
-                if (destpubkeys.size() > 2) {
+                if (destinations.size() > 2) {
                     CCerror = "no more than 2 dest pubkeys supported";
                     return NullUniValue;
                 }
             }
             if (V::EvalCode() == EVAL_TOKENSV2) {
-                if (destpubkeys.size() > 128) {
+                if (destinations.size() > 128) {
                     CCerror = "no more than 128 dest pubkeys supported";
                     return NullUniValue;
                 }
             }
 
-            mtx.vout.push_back(V::MakeTokensCCMofNvout(V::EvalCode(), 0, total, M, destpubkeys)); 
+            mtx.vout.push_back(V::MakeTokensCCMofNDestVout(V::EvalCode(), 0, total, M, destinations)); 
 
             // add optional custom probe conds to non-usual sign vins
             for (const auto &p : probeconds)
                 CCAddVintxCond(cp, p.first, p.second);
+
+            if (V::EvalCode() == EVAL_TOKENSV2) {
+                // probes for spending from mypk
+                for (const auto &mycond : GetTokenV2Conds(mypk))
```

Comment:
Could this `for` be moved to within the `if` below?

Reply:
I fixed all above except the boost mutex lock (I believe if we do this it is better to replace it everywhere, maybe next update)

We'd like to merge this PR and start testing with the GUI. Please if you have more notes, added them as an issue (if the branch is merged)